sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -711,21 +707,17 @@ object DateTimeUtils {
   def truncTimestamp(t: SQLTimestamp, level: Int, zoneId: ZoneId): SQLTimestamp = {
     level match {
       case TRUNC_TO_MICROSECOND => t
+      case TRUNC_TO_MILLISECOND =>
+        millisToMicros(microsToMillis(t))
```

Comment:
is it faster than `t - Math.floorMod(t, MICROS_PER_MILLI)`?

Reply:
I think it is almost the same:
- millisToMicros(microsToMillis(t)) ~= multiplyExact + floorDiv
- t - Math.floorMod(t, MICROS_PER_MILLI) = floorDiv  + * 
I can replaced it by floorMod for consistency

Reply:
Let's pick `t - Math.floorMod(t, MICROS_PER_MILLI)` for consistency.